tests/NHSD.BuyingCatalogue.Identity.Api.UnitTests/OrganisationControllerTests.cs

```diff
@@ -108,5 +108,41 @@ public async Task ShouldGetListOfOrganisations()
             organisationList[2].Name.Should().Be(org3.Name);
             organisationList[2].OdsCode.Should().Be(org3.OdsCode);
         }
+
+        [Test]
+        public async Task ShouldReturnNotFoundWhenNoOrgExists()
+        {
+            var orgMock = new Mock<IOrganisationRepository>();
+
+            OrganisationsController sut = new OrganisationsController(orgMock.Object);
+
+            var response = await sut.GetByIdAsync(Guid.NewGuid());
+
+            response.Should().BeEquivalentTo(new NotFoundResult());
+        }
+
+
+        [Test]
+        public async Task ShouldReturnOrganisationById()
+        {
+            var orgMock = new Mock<IOrganisationRepository>();
+            var expectedOrganisation = new Organisation(Guid.NewGuid(), "org name", "ods-code");
+            orgMock.Setup(x => x.GetByIdAsync(It.IsAny<Guid>()))
+                .ReturnsAsync(expectedOrganisation);
+
+            OrganisationsController sut = new OrganisationsController(orgMock.Object);
+
+            var result = await sut.GetByIdAsync(Guid.NewGuid());
+
+            result.Should().BeOfType<OkObjectResult>();
+            var objectResult = result as OkObjectResult;
+
+            objectResult.Value.Should().BeEquivalentTo(new OrganisationViewModel
+            {
+                OrganisationId = expectedOrganisation.Id,
+                Name = expectedOrganisation.Name,
+                OdsCode = expectedOrganisation.OdsCode
+            });
+        }
```

Comment:
Do you think we need a couple more tests?

One to ensure the dependencies are only called a certain number of times? e.g. GetByIdAsync is only called once on the IOrganisationRepository with the correct Organisation ID.

& the other to check what would happen if we got a NULL Organisation ID.